core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -511,44 +511,41 @@ public void clearDataMaps(AbsoluteTableIdentifier identifier) {
    */
   public void clearDataMaps(AbsoluteTableIdentifier identifier, boolean launchJob) {
     String tableUniqueName = identifier.getCarbonTableIdentifier().getTableUniqueName();
-    CarbonTable carbonTable = getCarbonTable(identifier);
-    if (launchJob && CarbonProperties.getInstance()
-        .isDistributedPruningEnabled(identifier.getDatabaseName(), identifier.getTableName())) {
+    if (launchJob) {
+      String jobClassName;
+      if (CarbonProperties.getInstance()
+          .isDistributedPruningEnabled(identifier.getDatabaseName(), identifier.getTableName())) {
+        jobClassName = DataMapUtil.DISTRIBUTED_JOB_NAME;
+      } else {
+        jobClassName = DataMapUtil.EMBEDDED_JOB_NAME;
+      }
+      // carbon table need to lookup only if launch job is set.
+      CarbonTable carbonTable = getCarbonTable(identifier);
       if (null != carbonTable) {
         try {
-          DataMapUtil.executeClearDataMapJob(carbonTable, DataMapUtil.DISTRIBUTED_JOB_NAME);
+          DataMapUtil.executeClearDataMapJob(carbonTable, jobClassName);
```

Comment:
Only when null != carbonTable, it will use jobClassName. So can you move 
`
 String jobClassName;
        .isDistributedPruningEnabled(identifier.getDatabaseName(), identifier.getTableName())) {	      if (CarbonProperties.getInstance()
          .isDistributedPruningEnabled(identifier.getDatabaseName(), identifier.getTableName())) {
        jobClassName = DataMapUtil.DISTRIBUTED_JOB_NAME;
      } else {
        jobClassName = DataMapUtil.EMBEDDED_JOB_NAME;
      }
`
into null != carbonTable?

Reply:
done